poleno/utils/tests/test_templatetags.py

```diff
@@ -241,18 +254,33 @@ def test_change_lang_tag(self):
         Checking that it generates correct links to the same view in different languages.
         """
         lang = ((u'de', u'Deutsch'), (u'en', u'English'), (u'fr', u'Francais'))
-        with self.settings(LANGUAGES=lang): # Fix active languages
-            r1 = self.client.get(u'/en/language/')
-            r2 = self.client.get(u'/de/language/')
-            r3 = self.client.get(u'/fr/language/')
-
-            for r in [r1, r2, r3]:
-                self.assertIs(type(r), HttpResponse)
-                self.assertEqual(r.status_code, 200)
-
-            self.assertEqual(r1.content, u'(/en/language/)(/de/language/)(/fr/language/)')
-            self.assertEqual(r2.content, u'(/en/language/)(/de/language/)(/fr/language/)')
-            self.assertEqual(r3.content, u'(/en/language/)(/de/language/)(/fr/language/)')
+        with self.settings(
+                LANGUAGES=lang,
+                MIDDLEWARE_CLASSES=[mc for mc in settings.MIDDLEWARE_CLASSES
+                                    if mc != u'django.middleware.locale.LocaleMiddleware'],
+                ):
+            for language, _ in lang:
+                with translation(language):
+                    r = self.client.get(u'/{}/language/'.format(language))
+                    self.assertIs(type(r), HttpResponse)
+                    self.assertEqual(r.status_code, 200)
+                    self.assertEqual(r.content, u'(/en/language/)(/de/language/)(/fr/language/)')
+
+    def test_change_lang_tag_with_missing_language_in_url(self):
+        u"""
+        Tests ``change_lang`` template tag by raising 404 exception, which is processed by custom
+        handler using a template with this tag.
```

Comment:
@viliambalaz V tom komentari je dolezite skor spomenut, ze sa testuje `change_lang` bez language prefixu v URL. Ze v takom pripade URL ostane bez prefixu pre vsetky jazyky. To ze to je testovane cez 404 exception nie je dolezite.

Reply:
:ok: